rfcs/0016-permissionless-assets-for-ibc-and-cw.md

```diff
@@ -0,0 +1,97 @@
+
+#### Non-sufficient assets implementation concerns
+
+#### Recieving Account doesnt hold pica during bridging
+**Potential solution**: relayer will send PICA to the reciever in case reciever doesnt have it. In case bridged asset is FlatFee asset, additional fee equal to ED in PICA will be charged from the transferred amount. Otherwise, percentage fee will be applied(To be configured)
+
+#### orml tokens pallet EDs
+Currently assets are being stored in orml tokens pallet which require Existential Deposits for stored assets. 
+
+**Potential Solution**: for non sufficient assets still provide non-zero Existential Deposit parameter which is similar to minimum balance in Asset Hub implementation. Then the Exitstential logic for orml tokens pallet will stay the same because account wont be able to hold a non sufficient asset without ED in parachain's native token. This will require a new field in AssetsRegistry: is_sufficient. All permissionless assets will have this field to be false by default. Assets created via permissioned extrinsic will allow to choose if asset is sufficient or not. Asset's sufficiency can be changed by governance.
+
+
+## Updated extrinsics
+
+Extrinsic to create asset via Root  `assets_registry.register_asset(name, symbol, decimals, location, ED, Ratio, is_sufficient)`.
+
+Extrinsic to create asset for bridging is `permissionless_assets.create_bridge_asset(name, symbol, decimals, location, ED)`.
+
+Extrinsic to create asset for CosmWasm is `permissionless_assets.create_cw_asset(name, symbol, decimals, ED)`.
+
+## permissionless_assets Storage Items
+
+#### For updating assets metadata
+
+`pub type Admins<T:Config> = StorageMap<_, Blake2_128Concat, T::AssetId, T::AccountId, OptionQuery>;`
+
+extrinsic `permissionless_assets.update_asset`  will check that caller of the extrinsic is an assets' admin. This storage item will be populated on `create_bridge_asset` extrinsic call.
+
+#### For minting/buring
+
+`pub type MintAdmin<T:Config> = StorageMap<_, Blake2_128Concat, T::AssetId, T::AccountId, OptionQuery>;`
+
+extrinsics `permissionless_assets.mint` and `permissionless_assets.burn` will check that caller of the extrinsic is an assets' admin and if so will use `AssetsTransactorRouter`'s `fungibles::Mutate` trait's `mint_into` or `burn_from` methods. This storage item will be populated on `create_cw_asset` extrinsic call.
+
+### Persmissionless IBC and XCM scenarios
+
+When assets are created by other chains over bridges,
+no symbol, no name, no suffiency(no ED) and no decimals defined.
+
+#### IBC ICS 20
+
+1. New token arrives.
+2. Relayer pays fee for creation of asset (with prefix).
+3. Account owner derived from client/connection/port/channel.
+4. Relayer pays fee for ED for account creation
+
+#### XCM
+
+1. New token arrives.
+2. There must be 2 tokens.
```

Comment:
for 2 tokens, do you mean 2 of the NSA token? like two separate transfers? or two different tokens? 

so if wants to register e.g BFC from bifrost, do they send 2 bfc transactions?

Reply:
user will sign one transaction.